Mathlib/RingTheory/Congruence/Basic.lean

```diff
@@ -111,6 +111,18 @@ protected theorem mul {w x y z} : c w x → c y z → c (w * y) (x * z) :=
   c.mul'
 #align ring_con.mul RingCon.mul
 
+protected theorem sub {S : Type*} [AddGroup S] [Mul S] (t : RingCon S)
+    {a b c d : S} (h : t a b) (h' : t c d) : t (a - c) (b - d) := t.toAddCon.sub h h'
```

Comment:
Does dot notation on `RingCon `not already work out of the box? by which I mean that without your new lemma I would expect `t.sub` to automatically expand to `t.toAddCon.sub`

Reply:
Ah maybe it proves `t.toAddCon (a - c) (b - d)`... Mind checking?

Reply:
The following gave an error
```lean
example  {S : Type*} [AddGroup S] [Mul S] (t : RingCon S) : true := by
  have := t.sub

/-
invalid field 'sub', the environment does not contain 'RingCon.sub'
  t
has type
  RingCon S
-/
```